rllib/algorithms/algorithm.py

```diff
@@ -576,65 +592,12 @@ def setup(self, config: PartialAlgorithmConfigDict):
 
         # Evaluation WorkerSet setup.
         # User would like to setup a separate evaluation worker set.
-
-        # Update with evaluation settings:
-        user_eval_config = copy.deepcopy(self.config["evaluation_config"])
-
-        # Merge user-provided eval config with the base config. This makes sure
-        # the eval config is always complete, no matter whether we have eval
-        # workers or perform evaluation on the (non-eval) local worker.
-        eval_config = merge_dicts(self.config, user_eval_config)
-        self.config["evaluation_config"] = eval_config
-
-        if self.config.get("evaluation_num_workers", 0) > 0 or self.config.get(
-            "evaluation_interval"
-        ):
-            logger.debug(f"Using evaluation_config: {user_eval_config}.")
-
+        if self.config.evaluation_num_workers > 0 or self.config.evaluation_interval:
             # Validate evaluation config.
-            self.validate_config(eval_config)
-
-            # Set the `in_evaluation` flag.
-            eval_config["in_evaluation"] = True
-
-            # Evaluation duration unit: episodes.
-            # Switch on `complete_episode` rollouts. Also, make sure
-            # rollout fragments are short so we never have more than one
-            # episode in one rollout.
-            if eval_config["evaluation_duration_unit"] == "episodes":
-                eval_config.update(
-                    {
-                        "batch_mode": "complete_episodes",
-                        "rollout_fragment_length": 1,
-                    }
-                )
-            # Evaluation duration unit: timesteps.
-            # - Set `batch_mode=truncate_episodes` so we don't perform rollouts
-            #   strictly along episode borders.
-            # Set `rollout_fragment_length` such that desired steps are divided
-            # equally amongst workers or - in "auto" duration mode - set it
-            # to a reasonably small number (10), such that a single `sample()`
-            # call doesn't take too much time and we can stop evaluation as soon
-            # as possible after the train step is completed.
-            else:
-                eval_config.update(
-                    {
-                        "batch_mode": "truncate_episodes",
-                        "rollout_fragment_length": 10
-                        if self.config["evaluation_duration"] == "auto"
-                        else int(
-                            math.ceil(
-                                self.config["evaluation_duration"]
-                                / (self.config["evaluation_num_workers"] or 1)
-                            )
-                        ),
-                    }
-                )
-
-            self.config["evaluation_config"] = eval_config
+            self.validate_config(self.config.evaluation_config)
```

Comment:
We do most of the config validation logic now inside the AlgorithmConfig class itself. So a lot of the validate_config code disappears. Eventually, we should get rid of validate_config entirely (should be handled by each config class directly).

Reply:
this is fantastic!